app/javascript/common/menu.js

```diff
@@ -0,0 +1,25 @@
+import http from './http';
+import { links } from '../react/config/config';
+
+const updateUrl = '/migration/menu_section_url?section=migration&url=/migration%23';
+const activeMenuItemSelector = '[data-target="#menu-migration"] > div > ul > li[class="list-group-item active"]';
+
+export const updateVerticalMenu = path => {
+  // POST /migration/menu_section_url to update session[:tab_url] on server
+  http.post(updateUrl + path);
```

Comment:
Looks like this is causing 

```
Server returned a non-200 response: undefined undefined
SyntaxError: Unexpected token h in JSON at position 1
angular.js:14961 TypeError: Cannot read property 'get' of undefined
    at ErrorModalController.$ctrl.show (error_modal_controller.self-...
```

because the `menu_section_url` method returns javascript, instead of json...

Except it shouldn't return anything, and the JS is the error handler code...

```js
$("#center_div").html("......Error text:\n\u003c/h2\u003e\n\u003cbr\u003e\n\u003ch3\u003e\nundefined method `parent_path\u0026#39; for nil:NilClass [migration/menu_section_url]...")
```

So, looks like something is not finding `parent_path` now, during `POST /migration/menu_section_url?section=migration&url=/migration%23/plans`

Reply:
For some reason I cannot reproduce this problem and a plain POST to that URL you showed above works correctly on my end.

Reply:
I think I know .. this only happens *after* rails autoreload, when it forgets it has any plugin menu sections

Reply:
Right. But then the question is whether we're actually discussing the problem in the right PR (since I don't know what we can possibly do about it in a plugin PR).

Reply:
Agreed, and it's not happening in production so maybe we don't have to treat it as a bug :).

(We could just make it not do anything when section is nil, but then we're just hiding errors, so idk, wdyt?)

Reply:
This whole thing should go away once we have a nice react-based menu. So I'd say we shouldn't do anything about it right now and just remove this logic once new menu lands.

Reply:
We already have the react menu...

Reply:
Anyway, it seems that with `updateVerticalMenu()` in place, the migration menu items stay highlighted (i.e. even when I move between migration screens). And withouth `updateVerticalMenu()` (i.e, without the POST) the menu item don't get highlighted at all.

Reply:
Ah, yes, `updateVerticalMenu` can't just set classes and expect it to work, it will have to change the react state instead of touching the html elements.

cc @Hyperkid123 , @fhlavac - do we have any JS way of telling the menu to change the active item?

Reply:
No, but we can use redux for it.